multiqc/modules/ngsderive/ngsderive.py

```diff
@@ -33,66 +33,51 @@ def __init__(self):
         self.readlen = {}
         self.encoding = {}
         self.junctions = {}
+        self.endedness = {}
 
         # parse ngsderive summary file
-        expected_header_count_strandedness = 5
-        expected_header_count_instrument = 4
-        expected_header_count_readlen = 4
-        expected_header_count_encoding = 3
-        expected_header_count_junctions = 9
-
         for f in self.find_log_files("ngsderive/strandedness"):
-            self.parse(
-                self.strandedness,
-                f,
-                "strandedness",
-                expected_header_count_strandedness,
-            )
+            self.parse(self.strandedness, f, "strandedness", expected_header_count=5)
 
         for f in self.find_log_files("ngsderive/instrument"):
-            self.parse(
-                self.instrument,
-                f,
-                "instrument",
-                expected_header_count_instrument,
-            )
+            self.parse(self.instrument, f, "instrument", expected_header_count=4)
 
         for f in self.find_log_files("ngsderive/readlen"):
-            self.parse(
-                self.readlen,
-                f,
-                "readlen",
-                expected_header_count_readlen,
-            )
+            self.parse(self.readlen, f, "readlen", expected_header_count=4)
 
         for f in self.find_log_files("ngsderive/encoding"):
-            self.parse(
-                self.encoding,
-                f,
-                "encoding",
-                expected_header_count_encoding,
-            )
+            self.parse(self.encoding, f, "encoding", expected_header_count=3)
 
         for f in self.find_log_files("ngsderive/junction_annotation"):
-            self.parse(
-                self.junctions,
-                f,
-                "junctions",
-                expected_header_count_junctions,
-            )
+            self.parse(self.junctions, f, "junctions", expected_header_count=9)
+
+        for f in self.find_log_files("ngsderive/endedness"):
+            self.parse(self.endedness, f, "endedness", expected_header_count=6)
+            self.parse(self.endedness, f, "endedness", expected_header_count=7)
```

Comment:
I take it that this is because the file could contain either 6 or 7 columns?

The problem with this is that we now get a warning log message for every sample:
```
|         ngsderive | Could not easily detect delimiter for file. Trying csv.Sniffer()
```
I think that we can drop this to debug at the best of times, but even so - I feel like a range could be a better approach?

Note that we only have a single test data file to work with here, so I can't be sure / test. It would be nice to have a few more example data files, covering different styles of output, if possible.

Reply:
This has been refactored to use a `set` of Ints.

And another example with a different style output uploaded: https://github.com/ewels/MultiQC_TestData/pull/269

Reply:
Looks like it actually fails with the new test, so it's great that we added it :D

Reply:
Pushed a fix